### PR TITLE
globus client setup

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made? 🤔

Setup for use of globus_client gem.  Split off from #2887 -- will rebase that PR when this is mergede.

## How was this change tested? 🤨

Existing tests

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -58,7 +58,7 @@ gem 'druid-tools'
 gem 'dry-types'
 gem 'edtf'
 gem 'faraday', '~> 2.0'
-gem 'globus_client', '~> 0.2'
+gem 'globus_client', '~> 0.3'
 gem 'honeybadger', '~> 4.0'
```
Comment (peetucket) [inline Gemfile:62]: we need at least 0.3 to ensure we have access to some new methods for checking user existence

**Review by mjgiarlo — APPROVED** [review]
Feel free to merge after removing the `require` statement. thanks!

File: config/application.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@
 require 'active_record/railtie'
 require 'active_storage/engine'
 # require 'rails/test_unit/railtie'
+require 'globus/client'
```
[the remark this answers is not in the document]
Comment (peetucket) [reply config/application.rb:15]: actually the new globus initializer failed when this wasn't present -- maybe i'm doing something wrong?

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply config/application.rb:15]: Bump to 0.3.1 and try again?

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**